Userland/Libraries/LibJS/Runtime/TypedArray.cpp

```diff
@@ -199,8 +258,15 @@ void TypedArrayBase::visit_edges(Visitor& visitor)
                 if (vm.exception())                                                                                                    \
                     return {};                                                                                                         \
             } else {                                                                                                                   \
-                /* FIXME: Initialize from Iterator or Array-like object */                                                             \
-                TODO();                                                                                                                \
+                auto iterator = first_argument.as_object().get(vm.well_known_symbol_iterator(), {}, true);                             \
+                if (iterator.is_function()) {                                                                                          \
+                    auto values = iterable_to_list(global_object(), first_argument, iterator);                                         \
```

Comment:
Missing exception check, this can throw

Reply:
the third argument is no side effects, so i assumed it doesnt produce any exceptions?

Reply:
oh, missed that - it shouldn't use the without_side_effects flag, that's not desired in most situations. If the object has an `@@iterator` key that has an accessor value, we want to call its getter.

So just `auto iterator = first_argument.as_object().get(vm.well_known_symbol_iterator());` is fine here.